### PR TITLE
fix: #6567 Drop image streams and use direct image references.  Also simplify operator config.yaml and CR by configuring full image names there.

### PR DESCRIPTION
**Pull request by chirino**
DevMode still uses image stream to pickup local builds.

### CODE REVIEW

File: install/operator/pkg/syndesis/action/install.go [hunk 1]
```diff
@@ -84,21 +83,17 @@ func (a *installAction) Execute(ctx context.Context, originalSyndesis *v1alpha1.
 	}
 	resourcesThatShouldExist[serviceAccount.GetUID()] = true
 
-	token, err := serviceaccount.GetServiceAccountToken(ctx, a.client, serviceAccount.Name, syndesis.Namespace)
```
Comment (lgarciaaco) [inline install/operator/pkg/syndesis/action/install.go]: I dont think oauth proxy will work if you remove this

File: install/operator/pkg/syndesis/action/upgrade.go [hunk 2]
```diff
@@ -45,11 +46,7 @@ func (a *upgradeAction) CanExecute(syndesis *v1alpha1.Syndesis) bool {
 
 func (a *upgradeAction) Execute(ctx context.Context, syndesis *v1alpha1.Syndesis) error {
 	if a.operatorVersion == "" {
-		operatorVersion, err := template.GetSyndesisVersionFromOperatorTemplate(a.scheme)
-		if err != nil {
-			return err
-		}
-		a.operatorVersion = operatorVersion
+		a.operatorVersion = pkg.DefaultOperatorTag
```
Comment (lgarciaaco) [inline install/operator/pkg/syndesis/action/upgrade.go]: Is this related with this PR?

**Comment by heiko-braun** [comment]
@lgarciaaco I wonder if this should be taken into 1.8.x

It might help with @asmigala ‘s request to pull from different registries,  no? What about the upgrade procedures you are working on, how is that impacted?

File: install/operator/tests/e2e/install_with_defaults_test.go [hunk 3]
```diff
@@ -17,240 +17,242 @@
...
+		imageName string
+		value     string
+	}{
+		{"Server", "syndesis-server", "1.8"},
+		{"Meta", "syndesis-meta", "1.8"},
+		{"S2I", "syndesis-s2i", "1.8"},
+		{"UI", "syndesis-ui", "1.8"},
+	}
+
+	syndesis.Spec.Components.Server.Image = flagtests[0].imageName + ":" + flagtests[0].value
```
Comment (lgarciaaco) [inline install/operator/tests/e2e/install_with_defaults_test.go]: It doesnt make much sense to test the tag change, but rather the whole image reference

**Comment by lgarciaaco** [comment]
> @lgarciaaco I wonder if this should be taken into 1.8.x
> 
> It might help with @asmigala ‘s request to pull from different registries, no? What about the upgrade procedures you are working on, how is that impacted?

I have a PR almost ready for the blockers ... which doesn't change the underlying structure, hence less risky.

This PR is doing some stuff that might impact install or upgrade, I have no idea. I agreed with @chirino we keep this in master, I fix the blockers,  we diverge master from 1.8.x and push to 1.8.x only bug fixes. That way @chirino can start bringing the new upgrade process to master and some other stuff he is probably working on. How does this sound?

File: install/operator/pkg/cmd/internal/install/install_standalone.go [hunk 4]
```diff
@@ -130,10 +127,10 @@ func (o *Install) installStandalone() error {
 		}
 
 		time.Sleep(2 * time.Second)
-		token, err := serviceaccount.GetServiceAccountToken(ctx, cli, "syndesis-oauth-client", syndesis.Namespace)
```
[the remark this answers is not in the document]
Comment (chirino) [reply install/operator/pkg/cmd/internal/install/install_standalone.go]: your right will fix this in a bit.

File: install/operator/pkg/generator/assets/addons/komodo/addon-komodo-server.yml.tmpl [hunk 5]
```diff
@@ -121,38 +125,31 @@
...
         from:
           kind: ImageStreamTag
-          name: {{ .Images.Syndesis.Komodo }}:{{ .Tags.Komodo }}
-          namespace: {{.Syndesis.Spec.ImageStreamNamespace}}
+          name: syndesis-dv:latest
+          namespace: {{.Syndesis.Namespace}}
       type: ImageChange
-    - type: ConfigChange
 
 - apiVersion: image.openshift.io/v1
```
[the remark this answers is not in the document]
Comment (chirino) [reply install/operator/pkg/generator/assets/addons/komodo/addon-komodo-server.yml.tmpl]: The S2I bits still use image streams.. Would need to change some java bits outside the operator to change this out.  Also need image streams when in --dev mode.

File: install/operator/pkg/syndesis/action/upgrade.go (hunk 2 above)
Comment (chirino) [reply install/operator/pkg/syndesis/action/upgrade.go]: We eliminated a bunch of variables from the conf.yaml, basically you don't get the operator version from the conf.yaml anymore.

File: install/operator/tests/e2e/install_with_defaults_test.go (hunk 3 above)
Comment (chirino) [reply install/operator/tests/e2e/install_with_defaults_test.go]: I've not been able to run the e2e tests..  :(

Comment (lgarciaaco) [reply install/operator/tests/e2e/install_with_defaults_test.go]: this is the command I use

```
GO111MODULE=on operator-sdk test local ./tests/e2e --debug --up-local --namespace syndesis-test --local-operator-flags "run --operator-config=build/conf/config.yaml" --go-test-flags "-v"
```

change the namespace

File: install/operator/pkg/syndesis/action/upgrade.go (hunk 2 above)
Comment (lgarciaaco) [reply install/operator/pkg/syndesis/action/upgrade.go]: ahhh ... ok check

File: install/operator/pkg/generator/assets/addons/komodo/addon-komodo-server.yml.tmpl (hunk 5 above)
[the remark this answers is not in the document]
Comment (lgarciaaco) [reply install/operator/pkg/generator/assets/addons/komodo/addon-komodo-server.yml.tmpl]: the idea with dev mode is to use the internally build docker image right? cant we point to that directly instead of to an imagetag?

**Comment by avano** [comment]
I tried to build the image from this PR, but ended up with:

```
{"level":"info","ts":1569487639.498213,"logger":"action","msg":"Installing Syndesis resource","type":"install","name":"example"}
{"level":"error","ts":1569487639.739823,"logger":"controller","msg":"Error reconciling","action":"*action.installAction","phase":"Installing","error":"service account token not found","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/syndesisio/syndesis/install/operator/pkg/controller/syndesis.(*ReconcileSyndesis).Reconcile\n\tsrc/github.com/syndesisio/syndesis/install/operator/pkg/controller/syndesis/syndesis_controller.go:120\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\nk8s.io/apimachinery/pkg/util/wait.Until\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
{"level":"error","ts":1569487639.7399592,"logger":"kubebuilder.controller","msg":"Reconciler error","controller":"syndesis-controller","request":"avano-test/example","error":"service account token not found","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/github.com/go-logr/zapr/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\nk8s.io/apimachinery/pkg/util/wait.Until\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
{"level":"info","ts":1569487640.7401538,"logger":"action","msg":"Installing Syndesis resource","type":"install","name":"example"}
{"level":"error","ts":1569487640.8690963,"logger":"controller","msg":"Error reconciling","action":"*action.installAction","phase":"Installing","error":"template: ./infrastructure/04-syndesis-oauth-proxy.yml.tmpl:64:32: executing \"./infrastructure/04-syndesis-oauth-proxy.yml.tmpl\" at <.OAuthClientSecret>: can't evaluate field OAuthClientSecret in type *generator.Context","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/syndesisio/syndesis/install/operator/pkg/controller/syndesis.(*ReconcileSyndesis).Reconcile\n\tsrc/github.com/syndesisio/syndesis/install/operator/pkg/controller/syndesis/syndesis_controller.go:120\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\nk8s.io/apimachinery/pkg/util/wait.Until\n\tsrc/github.com/syndesisio/syndesis/install/operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}

```

The service account was created, but the client wasn't substituted for some reason:

```
oc get sa syndesis-oauth-client                  
NAME                    SECRETS   AGE
syndesis-oauth-client   2         7m

```

**Comment by chirino** [comment]
rebased.
